lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -47,15 +47,25 @@ def capture(money, transaction_id, options={})
         end
       end
 
-      def purchase(money, card_or_token, options={})
-        commit('CreditSale') do |xml|
-          add_amount(xml, money)
-          add_allow_dup(xml)
-          add_customer_data(xml, card_or_token, options)
-          add_details(xml, options)
-          add_descriptor_name(xml, options)
-          add_payment(xml, card_or_token, options)
-          add_three_d_secure(xml, card_or_token, options)
+      def purchase(money, payment_method, options={})
+        if payment_method.is_a?(Check)
+          commit('CheckSale') do |xml|
+            add_check_payment(xml, payment_method, options)
+            add_amount(xml, money)
+            xml.hps :SECCode, 'WEB'
+            add_details(xml, options)
+            add_check_customer_data(xml, payment_method, options)
```

Comment:
Looking [at the docs](https://cert.api2.heartlandportico.com/Gateway/PorticoSOAPSchema/build/Default/webframe.html#Portico_xsd~e-PosRequest~e-Ver1.0~e-Transaction~e-CheckSale.html), the order of these methods should be switched.

Reply:
Possibly alarmingly, the tests seem to run either way, and HPS's test suite has them in the specified order. But the remote tests pass either way, so I've switched them.